### PR TITLE
feat: added refetch functionality for chart repo

### PR DESCRIPTION
**Pull request by pghildiyal**
# Description

Added functionality  to refetch charts from the chart repos

Fixes # devtron-labs/devtron#935

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?
This is tested by triggering functionality from the UI

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/chartRepo/chartRepo.service.tsx [hunk 1]
```diff
@@ -19,4 +19,9 @@ export const saveChartProviderConfig = (request: any, id: any) => {
 export const validateChartRepoConfiguration = (request: any):Promise<any> => {
     const URL = `${Routes.CHART_AVAILABLE}/${Routes.CHART_REPO}/validate`;
     return post(URL, request);
-  }
\ No newline at end of file
+}
+
+export const reSyncChartRepo = ():Promise<any> => {
+    const URL = `${Routes.CHART_AVAILABLE}/${Routes.CHART_REPO}/${Routes.CHART_RESYNC}`;
+    return get(URL);
```
Comment (manish-agrawal-ai) [inline src/components/chartRepo/chartRepo.service.tsx:26]: can be post (not fetching any data)

File: src/components/chartRepo/ChartRepo.tsx [hunk 2]
```diff
@@ -3,28 +3,59 @@ import { showError, useForm, useEffectAfterMount, useAsync, Progressing } from '
...
+    }
+
     return (
         <section className="global-configuration__component">
             <h2 className="form__title">Chart Repository</h2>
             <p className="form__subtitle">Manage your organization’s chart repositories.
             <span><a rel="noreferrer noopener" target="_blank" className="learn-more__href" href={DOCUMENTATION.GLOBAL_CONFIG_CHART}> Learn more</a> </span></p>
-            {[{ id: null, default: true, url: "", name: "", active: true, authMode: "ANONYMOUS" }].concat(result && Array.isArray(result.result) ? result.result : []).sort((a, b) => a.name.localeCompare(b.name)).map(chart => <CollapsedList {...chart} key={chart.id || Math.random().toString(36).substr(2, 5)} reload={reload} />)}
+            <CollapsedList  id={null} default={true} url={""} name={""} active={true} authMode={"ANONYMOUS"}  key={Math.random().toString(36).substr(2, 5)} reload={reload} />
+            <div className="chartRepo_form__subtitle float-left bold">Repositories({result.result.length})</div>
```
Comment (manish-agrawal-ai) [inline src/components/chartRepo/ChartRepo.tsx:44]: what if result is undefined - this code will fatal (checking at line no. 58)

**Review by manish-agrawal-ai — CHANGES_REQUESTED** [review]
please check comments

File: src/components/chartRepo/chartRepo.service.tsx (hunk 1 above)
Comment (pghildiyal) [reply src/components/chartRepo/chartRepo.service.tsx:26]: this needs to fixed at backend, ll have this and other changes done at the backend, post which will make changes here also

File: src/components/chartRepo/ChartRepo.tsx (hunk 2 above)
Comment (pghildiyal) [reply src/components/chartRepo/ChartRepo.tsx]: fixed

File: src/components/chartRepo/chartRepo.service.tsx (hunk 1 above)
Comment (manish-agrawal-ai) [reply src/components/chartRepo/chartRepo.service.tsx:26]: This is pending

**Review by manish-agrawal-ai — APPROVED** [review]
approved

**Pull request merged**